File: Sunrin-ICPC-2023/B/B.java
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int[] a = new int[n];
        for(int i=0; i<n; i++) a[i] = sc.nextInt();
        Arrays.sort(a);
        int res = 0;
        for(int i=0, j=n-1; i<j; i++){
            while(i < j && a[i] + a[j] > k) j--;
            if(i < j && a[i] + a[j] <= k){ res++; j--; }
        }
        System.out.println(res);
        sc.close();
    }
}

File: Sunrin-ICPC-2023/B/B.py
n, k = map(int, input().split())
a = list(map(int, input().split()))
a.sort()

i, j, res = 0, n - 1, 0
while i < j:
    while i < j and a[i] + a[j] > k: j -= 1
    if i < j and a[i] + a[j] <= k:
        res += 1
        j -= 1
    i += 1

print(res)